### DISCUSSION TITLE
Using ComputeSmallStrain in AbaqusUMATStress

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#20748] [Q&A Modules: Solid mechanics] [answered]
Hi all -

I'm looking through the Tensor Mechanics tests for the UMAT interface and they all appear to use a finite strain assumption. Furthermore, in looking at the [AbaqusUMATStress](https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C) source, it appears to be requesting a `strain_increment` material, among others, which are not available in the [ComputeSmallStrain](https://mooseframework.inl.gov/docs/doxygen/modules/classComputeSmallStrain.html) class.

So it appears the AbaqusUMATStress functionality is (currently) not compatible with a small strain approximation. Is this an Abaqus limitation? Or could AbaqusUMATStress be modified to also handle small strains?

**Comment by jessecarterMOOSE** [comment, marked as answer]
Tagging @dschwen @recuero since they worked on this.

**Reply by ngrilli** [reply]
Dear @jessecarterMOOSE 

Yes, I agree there is no way to switch to small strain,
however, I believe there is no need to do so.

You can use the small strain input variable DSTRAN and STRAN
to calculate the Cauchy stress inside the UMAT.

The concept of small/large strain in the UMAT depends on the specific choice of how the Cauchy stress is calculated,
if it is calculated with DSTRAN, STRAN, then you are using a small strain approximation,
if it is calculated with the the deformation gradients DFGRD0 and DFGRD1, then you are using a large strain formulation,
MOOSE is agnostic on the strategy you use to calculate the Cauchy stress, that is up to you,
MOOSE provides you with both small/large quantities and you can choose.

The use_displaced_mesh flag will be on by default,
therefore stress divergence calculation is made on the deformed mesh.

Best Regards,
Nicolò

**Reply by jessecarterMOOSE** [reply]
That makes sense and seems like a possible path, but from a programming standpoint, I already have a UMAT that works with Abaqus and I'd like to avoid making changes to make it compatible with MOOSE or making a MOOSE-specific version for different strain formulations.

**Reply by dschwen** [reply]
After a short discussion with @recuero I think we see what the issue is. Currently the `AbaqusUMATStress` material requires the material properties provided by our finite strain formulation to be present, even if the UMAT implements a small strain formulation. We can certainly fix that by adding a flag that makes the finite strain increments optional. 
Either way, it should not be necessary to have a custom UMAT with "changes to make it compatible with MOOSE".

**Reply by dschwen** [reply]
I'll go ahead and create an issue from this discussion.

**Reply by dschwen** [reply]
https://github.com/idaholab/moose/issues/20765

**Reply by dschwen** [reply]
@jessecarterMOOSE check out https://github.com/idaholab/moose/pull/20775

**Reply by jessecarterMOOSE** [reply]
Wow, I've been out of the loop a while. There are optional material properties now?

**Reply by dschwen** [reply]
Yep. I added then a little while ago. Hit me up if you want to know more about it.

But going back to the original question. Why don;t you just use _incremental small strain_, which should be fully compatible with our UMAT interface as is?

**Reply by jessecarterMOOSE** [reply]
Didn't realize that was an option. I'm curious to see if any of our Abaqus users would use that...
